metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java

```diff
@@ -279,30 +284,32 @@ public ResponseEntity<Object> getAspect(
             authentication,
             true);
 
-    return ResponseEntity.of(
-        buildEntityList(opContext, List.of(urn), Set.of(aspectName), withSystemMetadata).stream()
-            .findFirst()
-            .flatMap(
-                e ->
-                    e.getAspects().entrySet().stream()
-                        .filter(
-                            entry ->
-                                entry.getKey().equals(lookupAspectSpec(urn, aspectName).getName()))
-                        .map(Map.Entry::getValue)
-                        .findFirst()));
+    return buildEntityList(opContext, List.of(urn), Set.of(aspectName), withSystemMetadata).stream()
+        .findFirst()
+        .flatMap(
+            e ->
+                e.getAspects().entrySet().stream()
+                    .filter(
+                        entry -> entry.getKey().equals(lookupAspectSpec(urn, aspectName).getName()))
+                    .map(Map.Entry::getValue)
+                    .findFirst())
+        .map(ResponseEntity::ok)
+        .orElse(ResponseEntity.notFound().build());
```

Comment:
One of the things I don't like about 404s is that they're ambiguous about whether the resource is an invalid URI or data is missing. Can we add a response body or header indicating that this is just the urn is not found rather than the URI is invalid?